src/Core/Style.js

```diff
@@ -41,7 +41,7 @@ export function readExpression(property, ctx) {
             }
             return property.stops[0][1];
         } else if (property instanceof Function) {
-            return property(ctx.properties());
+            return property(ctx.properties(), ctx);
```

Comment:
Since properties are in the context, we could replace by `property(ctx)`; That would break the API though but  we could at least write a comment or open an issue to remember to do it at the next major itowns version.

Also, I think it should be documented (maybe in `StyleOptions`) that style options that are functions are called with these two parameters.

Reply:
I'm not so fond with breaking this API. We should do it with care since I'm sure this is used by multiple projects (maybe for a 3.x version).

Since there is still some developments on-going on `Style`, maybe we should wait for the stabilization of the API before documenting it?

Reply:
I'm ok with not breaking the API too. For this reason I feel like we should document it now since the API won't change (event if there is still work going on with `Style`). It may be done in #2196 or #2006, I will link this comment in one of these PRs so we don't forget about it.

I also think we should leave a comment in the code with the API breaking proposal to remember it when we'll prepare the 3.x version. Something starting with a keyword that we could re-use for other similar propositionns to easily find them all when we will need them. WDYT?

Reply:
We may document it in #2196 since we are moving and using the context for all converters. For the breaking proposal, we could either add a custom jsdoc tag or just a `TOBREAK` comment!